packages/scandipwa/src/component/ProductBundleItems/ProductBundleItems.container.js

```diff
@@ -73,42 +78,82 @@ export class ProductBundleItemsContainer extends ProductCustomizableOptionsConta
+                } = {}
+            } = {}
+        } = product;
+
+        const optionInitialPrice = itemPrice * quantity;
+        const optionPrice = finalItemPrice * quantity;
+        const optionPriceExclTax = finalItemPriceExclTax * quantity;
+        return {
+            optionPrice,
+            optionPriceExclTax,
+            optionInitialPrice
+        };
+    }
+
+    getOptionPrice(item, selectedValues, bundleDefaultPrices, isDynamicPrice) {
         const { option_id } = item;
         let price = 0;
         let priceExclTax = 0;
+        let initialPrice = 0;
 
         selectedValues.forEach(({ id, quantity, value }) => {
             if (option_id === id) {
                 const { options } = item;
 
-                options.forEach(({ id: optionId, product }) => {
+                options.forEach(({
+                    id: optionId, product, price: optionPrice, price_type: priceType
+                }) => {
                     if (JSON.stringify(value) === JSON.stringify([optionId.toString()])) {
                         if (product === null) {
                             return;
                         }
 
-                        const {
-                            price_range: {
-                                minimum_price: {
-                                    final_price: {
-                                        value: itemPrice = 0
-                                    } = {},
-                                    final_price_excl_tax: {
-                                        value: itemPriceExclTax = 0
-                                    } = {}
-                                } = {}
-                            } = {}
-                        } = product;
-
-                        price += itemPrice * quantity;
-                        priceExclTax += itemPriceExclTax * quantity;
+                        if (isDynamicPrice) {
+                            const {
+                                optionPrice,
+                                optionPriceExclTax,
+                                optionInitialPrice
+                            } = this.getDynanicOptionPrice(product, quantity);
+
+                            price += optionPrice;
+                            priceExclTax += optionPriceExclTax;
+                            initialPrice += optionInitialPrice;
+                        } else if (priceType === PRICE_TYPE_FIXED) {
+                            initialPrice += optionPrice * quantity;
+                            price += optionPrice * quantity;
+                            priceExclTax += optionPrice * quantity;
+                        } else if (priceType === PRICE_TYPE_PERCENT) {
+                            const {
+                                base_price: { value: defaultPrice } = 0,
+                                base_final_price: { value: defaultFinalPrice } = 0,
+                                base_final_price_excl_tax: { value: defaultFinalPriceExclTax } = 0
```

Comment:
This is wrong.
You expect `base_price`, `base_final_price` and `base_final_price_excl_tax` to be objects.
They cannot have default value zero.